src/moveit_visual_tools.cpp

```diff
@@ -96,48 +103,51 @@ bool MoveItVisualTools::loadPlanningSceneMonitor()
   // Check if we already have one
   if (psm_)
   {
-    ROS_WARN_STREAM_NAMED(LOGNAME, "Will not load a new planning scene monitor when one has "
+    RCLCPP_WARN_STREAM(LOGGER, "Will not load a new planning scene monitor when one has "
                                    "already been set for Visual Tools");
     return false;
   }
-  ROS_DEBUG_STREAM_NAMED(LOGNAME, "Loading planning scene monitor");
+  RCLCPP_INFO_STREAM(LOGGER, "Loading planning scene monitor");
 
   // Create tf transform buffer and listener
-  std::shared_ptr<tf2_ros::Buffer> tf_buffer = std::make_shared<tf2_ros::Buffer>();
-  std::shared_ptr<tf2_ros::TransformListener> tf_listener = std::make_shared<tf2_ros::TransformListener>(*tf_buffer);
+  // std::shared_ptr<tf2_ros::Buffer> tf_buffer = std::make_shared<tf2_ros::Buffer>();
+  // std::shared_ptr<tf2_ros::TransformListener> tf_listener = std::make_shared<tf2_ros::TransformListener>(*tf_buffer);
+
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer =
+      std::make_shared<tf2_ros::Buffer>(node_->get_clock(), tf2::durationFromSec(10.0));
+  std::shared_ptr<tf2_ros::TransformListener> tfl = std::make_shared<tf2_ros::TransformListener>(*tf_buffer);
 
   // Regular version b/c the other one causes problems with recognizing end effectors
-  psm_.reset(new planning_scene_monitor::PlanningSceneMonitor(ROBOT_DESCRIPTION, tf_buffer, "visual_tools_scene"));
+  psm_.reset(new planning_scene_monitor::PlanningSceneMonitor(node_, ROBOT_DESCRIPTION, tf_buffer, "visual_tools_scene"));
 
-  ros::spinOnce();
-  ros::Duration(0.1).sleep();
-  ros::spinOnce();
+  rclcpp::spin_some(node_);
```

Comment:
... and this... I don't think we should spin the node explicitly at all. This looks like a hack to work around some initialization issues. I would remove this and see if there are any problems. Most certainly, we can fix any syncing issues by waiting for the corresponding events using the PSM directly.

Reply:
You are right, I removed unnecessary spin_somes.